hooks/build

```diff
@@ -345,7 +398,12 @@ if [[ "${OSTYPE}" == "darwin"* ]]; then
 fi
 # Build id identifying the build uniquely
 BUILD_ID=${BUILD_ID:="local"}
-# Branch name for triggered builds
+# Branch name to build
+# Define an empty BRANCH_NAME_FIRST
+BRANCH_NAME=${BRANCH_NAME:=""}
+# in case SOURCE_BRANCH is defined it can override the BRANCH_NAME
+BRANCH_NAME=${SOURCE_BRANCH:=${BRANCH_NAME}}
+# Default branch name for triggered builds is master
 BRANCH_NAME=${BRANCH_NAME:="master"}
```

Comment:
Are the previous 3 statments the same as this?

```bash
BRANCH_NAME=${SOURCE_BRANCH:-"master"}
```

Reply:
Not really. You can override BRANCH_NAME in your configuration, but I can force it with SOURCE_BRANCH. I needed that mainly for cases before I merge it into master. For now I have to pretend that I am building from master branch even if I run it from ms-travis-ci-tests branch so that I can have it properly pushed to DockerHub registry. I configured DockerHub builds to have SOURCE_BRANCH="master". I will remove it during the merge to master. Those three lines will be changed to simply

`BRANCH_NAME=${BRANCH_NAME:="master"}`

and SOURCE_BRANCH will be removed completely

Reply:
👌